brian2/groups/neurongroup.py

```diff
@@ -111,10 +111,10 @@ def _get_refractory_code(self, run_namespace):
             variables = self.group.resolve_all(identifiers,
                                                run_namespace,
                                                user_identifiers=identifiers)
-            unit = parse_expression_unit(str(ref), variables)
-            if have_same_dimensions(unit, second):
+            dims = parse_expression_dimensions(str(ref), variables)
+            if dims is second.dim:
```

Comment:
Same as before. For example, might we not have something that evaluates as e.g. ``sqrt(second*second)`` which would fail ``is second.dim``?